py3status/modules/updates.py

```diff
@@ -0,0 +1,199 @@
+[{'full_text': 'Zypper '}, {'full_text': '34', 'color': '#ffa500'}]
+
+45pkgs
+[{'full_text': 'Xbps '}, {'full_text': '45', 'color': '#FF0000'}]
+
+no_updates
+{'full_text': 'No Updates'}
+"""
+
+
+class Update:
+    def __init__(self, parent, name, command):
+        self.parent = parent
+        self.name = name
+        self.command = command
+
+    def get_output(self):
+        try:
+            return self.parent.py3.command_output(self.command)
+        except self.parent.py3.CommandError:
+            return None
+
+    def count_updates(self, output):
+        return len(output.splitlines())
+
+    def get_updates(self):
+        output = self.get_output()
+        if output is None:
+            return {self.name: output}
+        return {self.name: self.count_updates(output)}
+
+
+class Apt(Update):
+    def count_updates(self, output):
+        return len(output.splitlines()[1:])
+
+
+class Apk(Update):
+    def count_updates(self, output):
+        return len(output.splitlines()[1:])
+
+
+class Eopkg(Update):
+    def get_output(self):
+        output = self.parent.py3.command_output(self.command)
+        if "No packages to upgrade." in output:
+            return ''
+        return output
+
+
+class Zypper(Update):
+    def count_updates(self, output):
+        return len([x for x in output.splitlines() if x][4:])
+
+
+class Py3status:
+    """
+    """
+
+    # available configuration parameters
+    cache_timeout = 600
+    format = None
+    thresholds = [(0, 'darkgray'), (10, 'degraded'),
+                  (20, 'orange'), (30, 'bad')]
+
+    class Meta:
+        deprecated = {
+            "rename_placeholder": [
+                {
+                    "placeholder": "aur",
+                    "new": "auracle",
+                    "format_strings": ["format"],
+                }
+            ]
+        }
+
+    def post_config_hook(self):
+        managers = [
+            ("pacman", "checkupdates"),
+            ("auracle", "auracle sync"),
+            ("yay", "yay --query --upgrades --aur"),
+            ("apk", "apk version -l '<'"),
+            ("apt", "apt list --upgradeable"),
+            ("eopkg", "eopkg list-upgrades"),
+            ("pkg", "pkg upgrade --dry-run --quiet"),
+            ("xbps", "xbps-install --update --dry-run"),
+            ("zypper", "zypper list-updates"),
+        ]
+
+        managed = getattr(self, 'managers', [])
+        if managed:
+            if isinstance(managed[0], tuple):
+                managers = managed
+            else:
+                new_managers = []
+                for name in managed:
+                    for manager in managers:
+                        if manager[0] == name:
+                            new_managers.append(manager)
+                            break
+                managers = new_managers
+        else:
+            placeholders = self.py3.get_placeholders_list(self.format or '')
+            managed = [x for x in placeholders if x != 'update']
```

Comment:
Can you describe what's happening in this `if-else` block? 

I think the logic here should be the following:

1. Parse format placeholders.
1. If it contains "updates", loop through all managers and determine which of them are available and use only them.
1. If it doesn't contain "updates", extract list of managers to use and use only them.

In other words, if my placeholder is `{pacman}`, even though I have `yay` installed I don't want this module to execute `yay` at all.

Reply:
> Figure out if it's a list of tuples or a list of manager names. From there, add them either via by direct entry or by a list of update_managers (above).

Support one way of configuring things, list of tuples is okay. No reason to have 10 ways to configure the same thing. If people want to use specific manager, they will do so via `format`.

> To answer your list...

I'm not sure it will work like I described, for example even if `managers` contains `yay`, if format doesn't contain it, no need to execute `yay` at all. Will it behave like this?

Reply:
In first place, users are not meant to use `managers`. It's a quick way for me to debug things via manipulating managers with a different list to see if everything works okay.

With no managers... If the `format` is empty, then we need to test everything in the `update_managers`. With `{yay}` in `format`, then it should only test `yay` in the `update_managers` and execute nothing but `yay`.

Basically, if your aur commands is generic and can be in `update_managers`, then I think `updates` should be able to detect this right away and execute only those. If your `format` is empty, then you might get more than what you want.. so make your own `format`.